UnityProject/Assets/Scripts/WaitFor.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+
+public static class WaitFor
+{
+	public static readonly WaitForEndOfFrame EndOfFrame = new WaitForEndOfFrame();
+	public static readonly WaitForFixedUpdate FixedUpdate = new WaitForFixedUpdate();
+	
+	private static Dictionary<float, WaitForSeconds> cachedWaitForSeconds = new Dictionary<float, WaitForSeconds>();
```

Comment:
Do you think these dictionary needs to implement a equality comparer to reliably use a float for the key? Due to floating point (im)precision, it may not memoize on computed values, e.g.:

```cs
(0.1f + 0.2f) == 0.3f // false
```

Could also consider having an `int` key and multiplying the float by 1000 (or whatever fraction of a second is the shortest reasonable wait).

Reply:
I don't think there should be a reason floating point accuracy should matter, it should return exactly the same `YieldInstruction` as if the constructor was used.

If they fail equality comparison, even if just do to float wonkiness then they should return a different `YieldInstruction`. If the floats fail equality, so too, should the `YieldInstruction` returned.
This is just my opinion of course. 

But I think rounding (which your suggestion essentially is) would just introduce unnecessary complexity with no benefit that I can currently see.

Of course, if I am misunderstanding something then please correct me.

Reply:
I think you understand it. I was concerned if there end up being a lot of waits that are calculated, you might see a "memory leak" where the cache is holding onto references to a bunch of yield instructions that are more-or-less identical to each other and rarely/never reused.